models/evc.py

```diff
@@ -1058,32 +1082,107 @@ def run_sdntrace(uni):
             return []
         return response.json().get('result', [])
 
-    def check_traces(self):
-        """Check if current_path is deployed comparing with SDN traces."""
-        trace_a = self.run_sdntrace(self.uni_a)
-        if len(trace_a) != len(self.current_path) + 1:
-            log.warning(f"Invalid trace from uni_a: {trace_a}")
-            return False
-        trace_z = self.run_sdntrace(self.uni_z)
-        if len(trace_z) != len(self.current_path) + 1:
-            log.warning(f"Invalid trace from uni_z: {trace_z}")
-            return False
-
-        for link, trace1, trace2 in zip(self.current_path,
-                                        trace_a[1:],
-                                        trace_z[:0:-1]):
-            if compare_endpoint_trace(
-               link.endpoint_a,
-               glom(link.metadata, 's_vlan.value'), trace2) is False:
-                log.warning(f"Invalid trace from uni_a: {trace_a}")
-                return False
-            if compare_endpoint_trace(
-               link.endpoint_b,
-               glom(link.metadata, 's_vlan.value'), trace1) is False:
-                log.warning(f"Invalid trace from uni_z: {trace_z}")
-                return False
+    @staticmethod
+    def run_bulk_sdntraces(uni_list):
+        """Run SDN traces on control plane starting from EVC UNIs."""
+        endpoint = f"{settings.SDN_TRACE_CP_URL}/traces"
+        data = []
+        for uni in uni_list:
+            data_uni = {
+                "trace": {
+                            "switch": {
+                                "dpid": uni.interface.switch.dpid,
+                                "in_port": uni.interface.port_number,
+                            }
+                        }
+                }
+            if uni.user_tag:
+                data_uni["trace"]["eth"] = {
+                                            "dl_type": 0x8100,
+                                            "dl_vlan": uni.user_tag.value,
+                                            }
+            data.append(data_uni)
+        response = requests.put(endpoint, json=data)
```

Comment:
Let's handle requests timeout to make sure it's prod grad, otherwise in worst cases this can hang. Probably 30 secs without retries should be fine, feel free to explore with other values if you wish. 

https://requests.readthedocs.io/en/latest/user/quickstart/#timeouts

It's OK not to have retries in this iteration since this will be periodic

Reply:
Done!

Reply:
You've added the timeout but you're not handling `requests.exceptions.Timeout`, so that only partly solves the issue, if a timeout happens it'll crash as it is. Make sure to handle the exception.

Reply:
Solved!